tests/suites/test_suite_ctr_drbg.function

```diff
@@ -51,50 +144,57 @@ exit:
 /* END_CASE */
 
 /* BEGIN_CASE */
-void ctr_drbg_validate_pr( data_t * add_init, data_t * entropy,
-                           data_t * add1, data_t * add2,
-                           data_t * result_str )
+void ctr_drbg_validate( int reseed_mode, char *nonce_string,
+                        int entropy_len_arg, char *entropy_string,
+                        char *reseed_string,
+                        char *add1_string, char *add2_string,
+                        char *result_string )
 {
-    mbedtls_ctr_drbg_context ctx;
-    unsigned char buf[512];
-
-    mbedtls_ctr_drbg_init( &ctx );
-
-
-    test_offset_idx = 0;
-    TEST_ASSERT( mbedtls_ctr_drbg_seed_entropy_len( &ctx, mbedtls_test_entropy_func, entropy->x, add_init->x, add_init->len, 32 ) == 0 );
-    mbedtls_ctr_drbg_set_prediction_resistance( &ctx, MBEDTLS_CTR_DRBG_PR_ON );
-
-    TEST_ASSERT( mbedtls_ctr_drbg_random_with_add( &ctx, buf, 16, add1->x, add1->len ) == 0 );
-    TEST_ASSERT( mbedtls_ctr_drbg_random_with_add( &ctx, buf, 16, add2->x, add2->len ) == 0 );
-    TEST_ASSERT( hexcmp( buf, result_str->x, 16, result_str->len ) == 0 );
-
-exit:
-    mbedtls_ctr_drbg_free( &ctx );
+    TEST_ASSERT( ctr_drbg_validate_internal( reseed_mode, nonce_string,
+                                             entropy_len_arg, entropy_string,
+                                             reseed_string, add1_string,
+                                             add2_string, result_string) == 0 );
 }
 /* END_CASE */
 
 /* BEGIN_CASE */
-void ctr_drbg_validate_nopr( data_t * add_init, data_t * entropy,
-                             data_t * add1, data_t * add_reseed,
-                             data_t * add2, data_t * result_str )
+void ctr_drbg_validate_no_reseed( char *add_init_string, char *entropy_string,
+                                  char *add1_string, char *add2_string,
+                                  char *result_string )
```

Comment:
The input data arguments are byte arrays, not null-terminated strings. They must have the type `data_t *` and you must use `foo->len`, not `strlen`.